src/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -25,7 +25,11 @@ internal static Boolean RequiresUnicode(String test)
 
             foreach (Char c in test)
             {
-                if (c > 127) return true;
+                // The Zip Format uses code page 437 when the Unicode bit is not set. This format
+                // is the same as ASCII for characters 32-126 but differs otherwise. If we can fit 
+                // the string into CP437 then we treat ASCII as acceptable.
+                if (c > 126 || c < 32)
```

Comment:
Not exactly, but it's functionally equivalent in most cases.

Different possible cases:
- If an entryName contains a character between 0-31, full .NET will throw an ArgumentException because it considers it an invalid path character. This check was removed a while back in https://github.com/dotnet/corefx/issues/4991 (on purpose) so whether we choose ASCII or UTF8 here is irrelevant from a compat perspective since either way we'll be different. To attain exact matching for those characters we would have to throw an ArgumentException which was agreed against in the aforementioned issue.
- If an entryName contains the character 127 we also have a difference as.NET 4.6 does not set the unicode bit in this case. However, [127 in CodePage 437 is not equivalent to ASCII 127](http://www-01.ibm.com/software/globalization/cp/cp00437.html) so writing it under ASCII and not setting the Unicode bit results in an invalid ZipFile. 
- The only other scenario I can think of in which they would differ is if the default system codepage was incompatible with ASCII between characters 32-127. On full .NET they would be written as non-Unicode using the default encoding but would be improperly read by other programs expecting that encoding to conform to CP437. Now they will just be encoded as ASCII as expected and work anywhere.

The main reason for the differences is that full framework encodes the non-unicode set using `Encoding.GetEncoding(0)`. [From the discussion]( https://github.com/dotnet/corefx/issues/8995) it was clear this was prone to creating non-conformant zips that would not be cross-platform or even compatible with other windows zip utilities.